### PR TITLE
Source Google Analytics (Universal Analytics): add start_date validation

### PR DESCRIPTION
**Pull request by arsenlosenko**
## What
Resolving this issue: https://github.com/airbytehq/oncall/issues/2034
User has ability to enter `start_date` in incorrect format during connector setup, sync is failing because of that

## How
Add `pattern` field to specification, and match this pattern either with valid date (`YYYY-MM-DD`) or empty string, as stated in the docs:
```
Enter the Replication Start Date in YYYY-MM-DD format. The data added on and after this date will be replicated. If this field is blank, Airbyte will replicate all data.
```
**Side note on regex in pattern**: last piece of it that matches every character is made to comply with CAT tests, namely the `test_validate_previous_configs` test which under the hood generates a bunch of fake random configs, thus in order to pass it's needed to add this part (if there is another option to skip this test feel free to add your suggestions) 

## ✅ User Impact ✅
This change takes backward compatibility into account, by including previously set patterns

### CODE REVIEW

**Comment by arsenlosenko** [comment]
canceled previous test run

**Review by davydov-d — APPROVED** [review]

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json [hunk 1]
```diff
@@ -81,7 +81,9 @@
         "type": "string",
         "title": "Replication Start Date",
         "description": "The date in the format YYYY-MM-DD. Any data before this date will not be replicated.",
-        "examples": ["2020-06-01"]
+        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$|^$|[\\s\\S]+$",
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/spec.json:84]: add unit tests please

**Review by lazebnyi — CHANGES_REQUESTED** [review]

**Review by lazebnyi — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  destination-cassandra test report (commit [`439b7affe2`](https://github.com/airbytehq/airbyte/pull/26296/commits/439b7affe2ee36996d4a315874e4cf511002d4c1)) - ❌

⏲️  Total pipeline duration: 07mn37s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-cassandra/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-cassandra docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-cassandra:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_2034-source-google-analytics-validate-replication-date/1688726422/439b7affe2ee36996d4a315874e4cf511002d4c1/destination-cassandra/0.1.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-cassandra test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-oracle-strict-encrypt test report (commit [`439b7affe2`](https://github.com/airbytehq/airbyte/pull/26296/commits/439b7affe2ee36996d4a315874e4cf511002d4c1)) - ❌

⏲️  Total pipeline duration: 10mn01s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-oracle-strict-encrypt/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-oracle-strict-encrypt docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-oracle:dev                                                       | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-oracle-strict-encrypt:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_2034-source-google-analytics-validate-replication-date/1688726422/439b7affe2ee36996d4a315874e4cf511002d4c1/destination-oracle-strict-encrypt/0.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-oracle-strict-encrypt test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-csv test report (commit [`439b7affe2`](https://github.com/airbytehq/airbyte/pull/26296/commits/439b7affe2ee36996d4a315874e4cf511002d4c1)) - ✅

⏲️  Total pipeline duration: 04mn18s 

| Step                                                                       | Result   |
|:---------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-csv/metadata.yaml     | ✅        |
| Connector version semver check                                             | ✅        |
| QA checks                                                                  | ✅        |
| Build connector tar                                                        | ✅        |
| Build destination-csv docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-csv:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_2034-source-google-analytics-validate-replication-date/1688726422/439b7affe2ee36996d4a315874e4cf511002d4c1/destination-csv/1.0.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-csv test
```

**Comment by octavia-squidington-iii** [comment]
##  source-google-analytics-v4 test report (commit [`439b7affe2`](https://github.com/airbytehq/airbyte/pull/26296/commits/439b7affe2ee36996d4a315874e4cf511002d4c1)) - ✅

⏲️  Total pipeline duration: 23mn28s 

| Step                                                                              | Result   |
|:----------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-google-analytics-v4/metadata.yaml | ✅        |
| Connector version semver check                                                    | ✅        |
| QA checks                                                                         | ✅        |
| Code format checks                                                                | ✅        |
| Connector package install                                                         | ✅        |
| Build source-google-analytics-v4 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                        | ✅        |
| Acceptance tests                                                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_2034-source-google-analytics-validate-replication-date/1688726422/439b7affe2ee36996d4a315874e4cf511002d4c1/source-google-analytics-v4/0.2.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-analytics-v4 test
```

**Comment by arsenlosenko** [comment]
fixed here https://github.com/airbytehq/airbyte/pull/28149

**Pull request closed without merging**